python/common/org/python/stdlib/datetime/timedelta.java

```diff
@@ -0,0 +1,183 @@
+    public timedelta(org.python.Object[] args, java.util.Map<java.lang.String, org.python.Object> kwargs) {
+        double days = 0;
+        double seconds = 0;
+        double microseconds = 0;
+
+        days += getArg(0, "days", args, kwargs);
+        seconds += getArg(1, "seconds", args, kwargs);
+        microseconds += getArg(2, "microseconds", args, kwargs);
+        microseconds += getArg(3, "milliseconds", args, kwargs) * 1000;
+        seconds += getArg(4, "minutes", args, kwargs) * 60;
+        seconds += getArg(5, "hours", args, kwargs) * 60 * 60;
+        days += getArg(6, "weeks", args, kwargs) * 7;
+
+        normalize(days, seconds, microseconds);
+    }
+
+    private void normalize(double days, double seconds, double microseconds) {
+        // convert decimals to smaller units e.g. 1.5 days => 1 days and 43200 seconds
+        seconds += (days % 1) * 60 * 60 * 24;
+        days -= days % 1;
+
+        microseconds += (seconds % 1) * 1e6;
+        seconds -= seconds % 1;
+
+        // round the answer to whole microseconds
+        microseconds = roundHalfToEven(microseconds);
+
+        // convert many microseconds to seconds
+        seconds += (long) (microseconds / 1e6);
+        microseconds -= ((long) (microseconds / 1e6)) * 1e6;
+        // convert many seconds to days
+        days += (long) (seconds / (24 * 60 * 60));
+        seconds -= (long) (seconds / (24 * 60 * 60)) * 24 * 60 * 60;
+
+        if (microseconds < 0) {
+            days -= 1;
+            seconds += MAX_SECONDS;
+            microseconds += MAX_MICROS + 1;
+
+            // IF we overflow MAX_SECONDS again
+            days += (long) (seconds / (24 * 60 * 60));
+            seconds -= (long) (seconds / (24 * 60 * 60)) * 24 * 60 * 60;
+        }
+        if (seconds < 0) {
+            days -= 1;
+            seconds += MAX_SECONDS + 1;
+        }
+
+        if (Math.abs(days) > MAX_DAYS) {
+            throw new org.python.exceptions.OverflowError(
+                    String.format("days=%d; must have magnitude <= %d", (long) days, MAX_DAYS));
+        }
+
+        this.days = org.python.types.Int.getInt((long) days);
+        this.seconds = org.python.types.Int.getInt((long) seconds);
+        this.microseconds = org.python.types.Int.getInt((long) microseconds);
+    }
+
+    private static long roundHalfToEven(double d) {
+        if (d % 0.5 == 0 && d % 1 != 0) {
+            if (Math.floor(d) % 2 == 0) {
+                return (long) Math.floor(d);
+            } else {
+                return (long) Math.ceil(d);
+            }
+        }
+        return Math.round(d);
+    }
+
+    @org.python.Method(__doc__ = "Return the total number of seconds contained in the duration.\n"
+            + " Equivalent to td / timedelta(seconds=1). For interval units \n"
+            + "other than seconds, use the division form directly \n"
+            + "(e.g. td / timedelta(microseconds=1)).", args = {})
```

Comment:
According to convention for `__doc__` should always end with '\n'
```suggestion
            + "(e.g. td / timedelta(microseconds=1)).\n", args = {})
```